crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -50,13 +66,22 @@ impl SharedObjectCongestionTracker {
     pub fn should_defer_due_to_object_congestion(
         &self,
         cert: &VerifiedExecutableTransaction,
-        max_accumulated_txn_cost_per_object_in_checkpoint: u64,
+        max_accumulated_txn_cost_per_object_in_checkpoint: Option<u64>,
         previously_deferred_tx_digests: &HashMap<TransactionDigest, DeferralKey>,
         commit_round: Round,
     ) -> Option<(DeferralKey, Vec<ObjectID>)> {
+        let tx_cost = match self.mode {
+            PerObjectCongestionControlMode::None => return None,
+            PerObjectCongestionControlMode::TotalGasBudget => cert.gas_budget(),
+            PerObjectCongestionControlMode::TotalTxCount => 1,
+        };
+
+        max_accumulated_txn_cost_per_object_in_checkpoint?;
+
         let shared_input_objects: Vec<_> = cert.shared_input_objects().collect();
         let start_cost = self.compute_tx_start_at_cost(&shared_input_objects);
-        if start_cost + cert.gas_budget() <= max_accumulated_txn_cost_per_object_in_checkpoint {
+
+        if start_cost + tx_cost <= max_accumulated_txn_cost_per_object_in_checkpoint.unwrap() {
```

Comment:
nit: instead of unwrap use `let Some(cost) = max_accumulated_txn_cost_per_object_in_checkpoint else { return None; }`

Reply:
Done